### DISCUSSION TITLE
Standalone API to build ModuleGraph of all entry points?

### DISCUSSION THREAD

**Discussion by romellem** [webpack/webpack#17328] [Q&A] [answered]
I use https://github.com/aackerman/circular-dependency-plugin to get information around circular dependencies.

The plugins [works by tapping into the `optimizeModules` hook](https://github.com/aackerman/circular-dependency-plugin/blob/21777bf980e78cb55bff0c91e43a03bbfb37f762/index.js#L23), which looks to be [the first hook that `modules` is ready](https://github.com/webpack/webpack/blob/56363993156e06a1230c9759eba277a22e6b6604/lib/Compilation.js#L2812). The plugin then recurses through the built `ModuleGraph` to find the circular dependencies.

However, running a full build seems to be a bit overkill just to get information around this `ModuleGraph`.

Is there a way to _only_ build this `ModuleGraph` without the overhead that comes with a full webpack compilation?

**Comment by alexander-akait** [comment, marked as answer]
Unfortunately, this is not entirely possible due to the loader logic, also some plugins can add more dependecies and modules

**Reply by romellem** [reply]
Ah that is right, I didn't think about new dependencies being injected from plugins.

OK, thank you! If there _was_ some "slimmer" API to build this graph, I would certainly find it useful, and I'd imagine others would as well. Or at least, a way to run the build process up until the `ModuleGraph` was fully built, and then have a way to bail from the build, that might also be a way to improve the speed of that reporting plugin.
